### PR TITLE
Halium 9.0 to 11.0 Catchup

### PR DESCRIPTION
**Pull request by fredldotme**
This is a continuation of PR #528 with a full rebase on top of libhybris upstream master.
Notable changes include:

- media compat layer enablement for Aethercast (Ubuntu Touch's remote display service)
- camera service enablement on Android 9.0 and up
- general compilation fixes

### CODE REVIEW

File: compat/media/Android.mk [hunk 1]
```diff
@@ -67,12 +91,25 @@ include $(BUILD_EXECUTABLE)
...
+# -------------------------------------------------
+
 include $(CLEAR_VARS)
 include $(LOCAL_PATH)/../Android.common.mk
 
 HYBRIS_PATH := $(LOCAL_PATH)/../../hybris
 
-LOCAL_CFLAGS += -std=gnu++0x
+ifneq ($(IS_ANDROID_8),true)
+    LOCAL_CFLAGS += -std=gnu++0x
```
Comment (mlehtima) [inline compat/media/Android.mk:111]: Some indentation mismatches in this file here and other places below..

File: compat/surface_flinger/surface_flinger_compatibility_layer.cpp [hunk 2]
```diff
@@ -330,13 +332,23 @@ SfSurface* sf_surface_create(SfClient* client, SfSurfaceCreationParameters* para
...
 
-	sf_client_begin_transaction(client);
-	{
-		surface->surface_control->setPosition(params->x, params->y);
-		surface->surface_control->setLayer(params->layer);
-		surface->surface_control->setAlpha(params->alpha);
-	}
-	sf_client_end_transaction(client);
+#if ANDROID_VERSION_MAJOR>=8
+    uint32_t mDisplayLayerStack = 0;
```
Comment (mlehtima) [inline compat/surface_flinger/surface_flinger_compatibility_layer.cpp:336]: Indentation issues (spaces vs tabs) here and many places below. This file seems to be using tabs.

File: compat/camera/Android.mk [hunk 3]
```diff
@@ -25,9 +31,16 @@ LOCAL_MODULE_TAGS := optional
 LOCAL_C_INCLUDES := \
 	$(HYBRIS_PATH)/include
 
+ifeq ($(IS_ANDROID_8),true)
+LOCAL_CFLAGS += \
+    -Wno-unused-parameter \
```
Comment (mlehtima) [inline compat/camera/Android.mk:36]: Indentation should use tabs.

**Review by mlehtima — CHANGES_REQUESTED** [review]
I found some style issues when doing a quick check of the changes. Not sure if I found all. I will have more thorough look tomorrow.

**Comment by mlehtima** [comment]
@fredldotme do you have any plans to fix the style issues I found or should I fix them after merging?

**Comment by fredldotme** [comment]
> @fredldotme do you have any plans to fix the style issues I found or should I fix them after merging?

Yes please, I would very much welcome someone helping me. I'm *drowning* in things to patch. :)

**Pull request merged**